src/service/triple-store-service.js

```diff
@@ -71,38 +70,76 @@ class TripleStoreService {
         //     knowledgeAssetsUALs.map((ual) => `<${ual}>`),
         // );
         // const unifiedGraphTriples = [...triples, ...tripleAnnotations];
+        const publicKnowledgeAssetsTriples = this.dataService.groupTriplesBySubject(
+            triples.public ?? triples,
+        );
 
         const promises = [];
+        if (triples.private && triples.private.length !== 0 && !existsInNamedGraphs) {
+            const privateKnowledgeAssetsTriples = this.dataService.groupTriplesBySubject(
+                triples.private,
+            );
+            const privateKnowledgeAssetsStatesUALs = [];
+            let privateSubject;
+            let publicSubject;
+            let publicIndex = 0;
+            let privateIndex = 0;
+            while (
```

Comment:
This should be done through assertion-tools in dataService. Parsing should be done using N3.js

Reply:
This parsing won't work for all cases and also not optimal way of doing it

Reply:
@brkagithub  will create PR which will fix this